### DISCUSSION TITLE
Multi-tenant engine config

### DISCUSSION THREAD

**Discussion by quom** [litestar-org/advanced-alchemy#260] [Q&A]
Currently my app is set up with one database per tenant (plus a catalog db for storing metadata about tenants: db server, db name etc) and so fetching the db engine requires request state to determine what tenant db to use (e.g. hostname, header, path variable). 

Is there a way for this to be achieved with advanced alchemy currently? I can't see any way of doing this without creating one `StarletteAdvancedAlchemy` instance per db engine - would this work as intended?

**Comment by provinzkraut** [comment]
I think you should be able to do something if you overwrite the `get_session` method: 
https://github.com/litestar-org/advanced-alchemy/blob/be8054e394c9f369d9fa460779e914fd18cab67c/advanced_alchemy/extensions/starlette.py#L267

It has access to the `Request` instance, so you should be able to extract the info you need there and select an engine based on it.

```python

    def get_session(self, request: Request) -> SessionT:
        session = getattr(request.state, self.session_key, None)
        if session is not None:
            return cast(SessionT, session)

        engine_mapping = request.app.state["tenant_engines"]
        engine = request.headers["tenant_id"]
  
        session = AsyncSession(bind=engine)
        setattr(request.state, self.session_key, session)
        return session
```

**Comment by Alc-Alc** [comment]
Multi config is now supported. You can pass in a list of config objects when you create the extension.

https://github.com/litestar-org/advanced-alchemy/blob/main/advanced_alchemy/extensions/starlette/extension.py#L36
